src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/model/Prisoner.kt

```diff
@@ -1,23 +1,29 @@
 package uk.gov.justice.digital.hmpps.prisonersearch.model
 
 import io.swagger.v3.oas.annotations.media.Schema
+import org.apache.commons.lang3.builder.DiffResult
+import org.apache.commons.lang3.builder.Diffable
 import org.springframework.data.annotation.Id
 import org.springframework.data.elasticsearch.annotations.DateFormat
 import org.springframework.data.elasticsearch.annotations.Document
 import org.springframework.data.elasticsearch.annotations.Field
 import org.springframework.data.elasticsearch.annotations.FieldType
 import org.springframework.data.elasticsearch.annotations.InnerField
 import org.springframework.data.elasticsearch.annotations.MultiField
+import uk.gov.justice.digital.hmpps.prisonersearch.services.diff.DiffableProperty
+import uk.gov.justice.digital.hmpps.prisonersearch.services.diff.PropertyType
+import uk.gov.justice.digital.hmpps.prisonersearch.services.diff.getDiffResult
 import java.time.LocalDate
 
-open class Prisoner {
+open class Prisoner : Diffable<Prisoner> {
```

Comment:
Currently using the Apache Commons difference engine. [JaVers](https://javers.org/) looks to be a much more sophisticated difference engine with Spring integration - something to consider if we start getting more complicated.